### ISSUE TITLE
code to fine-tune

### ISSUE THREAD

**Issue by dmoham1476** [openai/CLIP#21]
How can I fine-tune using model.py with my custom dataset and the pre-trained model.pt?

**Comment by jongwook** [comment]
We don't currently plan to release training or fine-tuning examples in this repo, but in general fine-tuning is done with:

- Load the pre-trained model, usually with a prediction "head" for your fine-tuning task.
- Initialize an optimizer, with an appropriate learning rate which you'll need to tune.
- Repeat: feed the data through the model (and the head) and update your model to minimize your task-specific loss.
- Stop the updates after a fixed number of iterations or when your early stopping criterion is met.

**Comment by cupcakefromchina** [comment]
> * Load the pre-trained model, usually with a prediction "head" for your fine-tuning task.
> * Initialize an optimizer, with an appropriate learning rate which you'll need to tune.
> * Repeat: feed the data through the model (and the head) and update your model to minimize your task-specific loss.
> * Stop the updates after a fixed number of iterations or when your early stopping criterion is met.

May I ask some implement details about the finetune on ImageNet？
In the page 14, you write as below :
"Although adapting CLIP to the ImageNet distribution increases its ImageNet accuracy by 9.2% to 85.4% overall, and ties the accuracy of the 2018 SOTA from Mahajan et al. (2018), average accuracy under distribution shift slightly decreases"

In this version of model, the text encoder is still working or not working? It seems that you only use the clip features which extracted by the image encoder.

Thank you for your work.

**Comment by iremonur** [comment]
> We don't currently plan to release training or fine-tuning examples in this repo, but in general fine-tuning is done with:
> 
> * Load the pre-trained model, usually with a prediction "head" for your fine-tuning task.
> * Initialize an optimizer, with an appropriate learning rate which you'll need to tune.
> * Repeat: feed the data through the model (and the head) and update your model to minimize your task-specific loss.
> * Stop the updates after a fixed number of iterations or when your early stopping criterion is met.

What do you mean by "with a prediction "head"?  As I understood, CLIP encodes images and texts with its backbone network and then the feature maps are compared by cosine similarity, and the distance between image-text pairs is tried to reduce by the back-propagation algorithm. So, CLIP does not make a classification, it aims to reduce the distance between ground-truth image-text pairs regardless of their class. So, is there a prediction head in the original CLIP, if not so why I should add to fine-tune it with my own data?
